cfme/services/requests.py

```diff
@@ -218,69 +66,270 @@ def find_request(cells, partial_check=False):
             logger.debug(' Request Message: %s', row.last_message.text)
             return row
     else:
-        # Request not found at all, can't continue
+        raise Exception("The requst specified by {} not found!".format(str(cells)))
         return False
 
 
-def go_to_request(cells, partial_check=False):
-    """Finds the request and opens the page
-
-    See :py:func:`wait_for_request` for futher details.
+# TODO Refactor these module methods and their callers for a proper request class
+class Request(Navigatable):
+    def __init__(self, row_description=None, cells=None, partial_check=False, appliance=None):
+        Navigatable.__init__(self, appliance=appliance)
+        self.row_description = row_description
+        self.cells = {'Description': self.row_description} if not cells else cells
+        self.partial_check = partial_check
+        self.REQUEST_FINISHED_STATES = {'Migrated', 'Finished'}
+        self.row = find_request(self.cells, self.partial_check)
+        self.request_id = self.row.request_id.text
+        self.description = self.row.description.text
+
+    @property
+    def if_exists(self):
+        return True if find_request(cells=self.cells, partial_check=self.partial_check) else False
+
+    def load_details(self):
+        return navigate_to(self, 'Details')
+
+    def reload(self):
+        toolbar.refresh()
+
+    def update_row(self):
+        self.row = find_request(self.cells, self.partial_check)
+
+    def approve_request(self, reason, cancel=False):
```

Comment:
left as it was before - maybe it would be better to leave only class method and remove view method.